ivy.el

```diff
@@ -1661,7 +1670,9 @@ customizations apply to the current completion session."
                                     collection))))
         (ivy-display-function
          (unless (window-minibuffer-p)
-           (cdr (assoc caller ivy-display-functions-alist)))))
+           (or ivy-display-function
+               (cdr (or (assq caller ivy-display-functions-alist)
+                        (assq t ivy-display-functions-alist)))))))
```

Comment:
Doesn't the first operand in `(or ivy-display-function ...)` make it impossible to customise `ivy-display-function` on a per-command basis?

Reply:
If user want to config per-command function, they should use  ivy-display-functions-alist and set ivy-display-function to nil.

Reply:
maybe we should doc this:-)

Reply:
That is a bad idea for two important reasons:
1. It is a breaking change.
2. It makes `ivy-display-function` redundant for any setting other than `nil`, as the user will have to customise `ivy-display-functions-alist` for *every single* command they use.

Reply:
On further inspection, I'm not quite sure what the intended interplay between `ivy-display-function` and `ivy-display-functions-alist` is at the moment, so I retract my previous and defer to @abo-abo for comment.

Some comments/questions/observations:
* `ivy-display-function` is also used in `ivy--insert-minibuffer`.
* Why do we check for `window-minibuffer-p` before `let`-binding `ivy-display-function`? Could this be simplified as a lookup into `ivy-display-functions-alist`, in which `ivy-display-function` is mapped to by the fallback key `t`?
* If `window-minibuffer-p` is necessary, shouldn't it be `(minibuffer-window-active-p (selected-window))` instead, seeing as there can be multiple minibuffer windows?

Reply:
maybe ivy-display-functions-alist should override ivy-display-function :-)

Reply:
We check `window-minibuffer-p` because the only viable display function (other than `nil`) at the time was `ivy-display-function-overlay`, which doesn't make sense to use when you press `TAB` while in the minibuffer.